### DISCUSSION TITLE
Morphometric Calculator: FAI calculation on non rectangular grid

### DISCUSSION THREAD

**Discussion by jiangli2036** [UMEP-dev/UMEP#382] [Q&A]
Hello,

I have technical questions when doing calculation on non rectangular grids.
I want to run the morphometric calculator for the buildings DSM as seen in the attached first image for the area denoted by the red polygon (inside the blue bounding box).  The second image is the setting page where I have tried to set the search distance to be 150m, 42m, 25m from the grid centriod. They all successed, but I am very confused about the results. I also took screenshots of the beginning part of each of the three results below. So, My questions are:

1.What is the calculation principle (specific formula or figure?) of this part? I am concerned about the Frontal Area Index, e.g. how to define the windward frontal area and ground area of the building under each search direction? 

2. If I want to get the FAI of this non rectangular polygon, how long a search distance would be appropriate? what should I consider when choosing a search distance?

3. How can I calculate the FAI on a specific wind direction, e.g. southeast wind?

Your answers would be greatly helpful for my research! Thanks a lot!

QGIS version : 3.24.0
UMEP version : 3.19.2
<img width="282" alt="2" src="https://user-images.githubusercontent.com/64188938/159160934-dcb50cd6-02f0-42ec-a037-87ed005a00f7.png">
<img width="557" alt="1" src="https://user-images.githubusercontent.com/64188938/159160951-00659c87-da28-4898-bbfc-274d346c75b6.png">
![image](https://user-images.githubusercontent.com/64188938/159161057-ca34468d-89e5-4e79-83e5-15dbe29136d4.png)

**Comment by biglimp** [comment]
The Morphometric Calculator takes the bounding box of the polygon and calculates the FAI etc. This means that a strangely shaped polygon might be you strange values. Also, diagonal polygons (which I think I see in your case) also includes areas outside since the bouning box is a measure of min and max coordinate from the four cardinal directions (N, E, S, W). The search distance to include a non-regular polygon  depends on the shape but you have to make some simplifications to include "as much of the polygon without including too much areas outside.

With this said, this tool is maybe more appropriate on a neighborhood (200-1000m) scale and it looks like you are trying to use it in a bit too fine scale.

If you like FAI from a specific wind direction I would recommend to take i "pie"-shaped aggregated value from the text files produced, For example, if you would like to know the FAI from south I would take a mean from maybe 165 to 195 degrees.

**Reply by jiangli2036** [reply]
Many thanks! Can I rotate the map to let the polygon overlap its bounding box as much as possible, then, use the radius of the shorter side of the bounding box as the search distance if I start search from the centroid?

**Reply by biglimp** [reply]
Yes, you could. However, geographically, maybe not the best idea but if you are just interested in small areas it should be ok...

**Reply by jiangli2036** [reply]
Yes, because I don't want to lose the scale of a real city block, thank you for your answer.

**Reply by PetitMote** [reply]
Hi, is it ok if I react to this here ? I’d like to know why the scale is important. Is it because the algorithm is imprecise, or is it because of some other technical specificities ?
